error.go

```diff
@@ -186,6 +194,25 @@ func (e *Error) SetTransaction(tx *Transaction) {
 	tx.mu.RUnlock()
 }
 
+// Cause returns original error assigned to Error, nil if Error or Error.cause is nil.
+// https://godoc.org/github.com/pkg/errors#Cause
+func (e *Error) Cause() error {
+	if e != nil {
+		return e.cause
+	}
+	return nil
+}
+
+// Error returns string message for error.
+// if Error or Error.cause is nil, "[EMPTY]" will be used.
```

Comment:
I think for log-type errors, the log message should be returned by `Error()`, while the associated error is returned by `Cause()` (as you do already.)

Perhaps change NewError and NewErrorLog to set another error message field at construction time, alongside the cause?

Reply:
Hmm, you mean I should add another `error` field to both `ErrorLogRecord` and `ErrorData` structures?
Not sure when should I use them. Could you explain?

Reply:
I mean add an error message field to `Error`, so it looks something like:

```go
type Error struct {
    *ErrorData
    cause error
    err string
}

func (e *Error) Error() string {
    return e.err
}

func (e *Error) Cause() error {
    return e.cause
}
```

`NewErrorLog` would set the ErrorLogRecord.Message value, and `NewError` would just copy across the value from the provided error.

What do you think?

Reply:
Good idea - I just wonder about scenario when `NewErrorLog` is called with nil error. Do you think it is possible on production? 
This scenario exists in `validation_test.go:236 TestValidateErrorLog` and not sure should I change test to have non-nil error value or not.
At the moment I bypassed problem by settings `Error.err` manually afterwards but I really don't like this solution.

Reply:
Thank you for detailed explanation - I revert changes made to newError function and initialize fields later.